src/startupscriptgenerator/dotnetcore/scriptgenerator.go

```diff
@@ -24,59 +20,52 @@ type DotnetCoreStartupScriptGenerator struct {
 	Manifest           common.BuildManifest
 }
 
-type projectDetails struct {
-	Name         string
-	FullPath     string
-	Directory    string
-	CSProjectObj csProject
-}
-
-// Object models representing .NET Core '.csproj' file xml content
-type csProject struct {
-	SdkName    string          `xml:"Project,Sdk"`
-	Properties []propertyGroup `xml:"PropertyGroup"`
-}
-
-type propertyGroup struct {
-	AssemblyName string `xml:"AssemblyName"`
-}
-
-const ProjectEnvironmentVariableName = "PROJECT"
 const DefaultBindPort = "8080"
-
-var _projDetails projectDetails = projectDetails{}
+const RuntimeConfigJsonExtension = "runtimeconfig.json"
 
 func (gen *DotnetCoreStartupScriptGenerator) GenerateEntrypointScript(scriptBuilder *strings.Builder) string {
 	logger := common.GetLogger("dotnetcore.scriptgenerator.GenerateEntrypointScript")
 	defer logger.Shutdown()
 
-	logger.LogInformation(
-		"Generating script for source at '%s' and published output at '%s'",
-		gen.SourcePath,
-		gen.AppPath)
-
-	startupDllFileName := gen.getStartupDllFileName()
+	logger.LogInformation("Generating script for published output at '%s'", gen.AppPath)
 
 	// Expose the port so that a custom command can use it if needed
 	common.SetEnvironmentVariableInScript(scriptBuilder, "PORT", gen.BindPort, DefaultBindPort)
 	scriptBuilder.WriteString("export ASPNETCORE_URLS=http://*:$PORT\n\n")
 
+	defaultAppFileDir := filepath.Dir(gen.DefaultAppFilePath)
+
 	scriptBuilder.WriteString("readonly appPath=\"" + gen.RunFromPath + "\"\n")
 	scriptBuilder.WriteString("userStartUpCommand=\"" + gen.UserStartupCommand + "\"\n")
 	scriptBuilder.WriteString("startUpCommand=\"\"\n")
-	scriptBuilder.WriteString("readonly startupDllFileName=\"" + startupDllFileName + "\"\n")
+	scriptBuilder.WriteString("readonly defaultAppFileDir=\"" + defaultAppFileDir + "\"\n")
 	scriptBuilder.WriteString("readonly defaultAppFilePath=\"" + gen.DefaultAppFilePath + "\"\n")
+	scriptBuilder.WriteString("echo Finding the startup file name...\n")
+	scriptBuilder.WriteString("cd \"$appPath\"\n")
+	scriptBuilder.WriteString("for file in *; do \n")
+	scriptBuilder.WriteString("  if [ -f \"$file\" ]; then \n")
+	scriptBuilder.WriteString("    case $file in\n")
+	scriptBuilder.WriteString("      *.runtimeconfig.json)\n")
+	scriptBuilder.WriteString("        startupDllFileNamePrefix=${file%%.runtimeconfig.json}\n")
+	scriptBuilder.WriteString("        startupDllFileName=\"$startupDllFileNamePrefix.dll\"\n")
+	scriptBuilder.WriteString("        echo \"Found the startup file '$startupDllFileName'\"\n")
+	scriptBuilder.WriteString("        break\n")
+	scriptBuilder.WriteString("      ;;\n")
+	scriptBuilder.WriteString("    esac\n")
+	scriptBuilder.WriteString("  fi\n")
+	scriptBuilder.WriteString("done\n")
 	scriptBuilder.WriteString("if [ ! -z \"$userStartUpCommand\" ]; then\n")
 	scriptBuilder.WriteString("  cd \"$appPath\"\n")
 	scriptBuilder.WriteString("  startUpCommand=\"$userStartUpCommand\"\n")
```

Comment:
for the other languages, we typically check for the case where the passed in command is not really a command, but the path to the main file (e.g. `server.js`, `app.py`, etc) - in this case the dll file. If the file exists and ends with `.dll` we could just prepend dotnet to it.

Reply:
I feel it is a bit different for dotnet case though...here we would have one designated (via runtimeconfig.json) dll as startup and where as in case of other languages, it is a source repo with bunch of files and there is no standard way one can know which is the startup file and providing that support for just the file probably makes sense, but I think we need not worry about that case here IMO. We could revisit this later based on feedback.